com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/scrbound.compute

```diff
@@ -1,538 +1,728 @@
-    float4x4 g_mProjection = g_mProjectionArr[eyeIndex];
+    M._14_24_34 = d; // Last column
 
-    //uint vindex = groupID * NR_THREADS + threadID;
-    unsigned int g = groupID;
-    unsigned int t = threadID;
+    return M;
+}
 
-    const int subLigt = (int) (t/8);
-    const int lgtIndex = subLigt+(int) g*8;
-    const int sideIndex = (int) (t%8);
+// Scale followed by rotation (scaled axes).
+float3x3 ScaledRotation3x3(float3 xAxis, float3 yAxis, float3 zAxis)
+{
+    float3x3 R = float3x3(xAxis, yAxis, zAxis);
+    float3x3 C = transpose(R); // Row to column
 
-    const int eyeAdjustedLgtIndex = GenerateLightCullDataIndex(lgtIndex, g_iNrVisibLights, eyeIndex);
-    SFiniteLightBound lgtDat = g_data[eyeAdjustedLgtIndex];
+    return C;
+}
 
-    const float3 boxX = lgtDat.boxAxisX.xyz;
-    const float3 boxY = lgtDat.boxAxisY.xyz;
-    const float3 boxZ = -lgtDat.boxAxisZ.xyz;           // flip axis (so it points away from the light direction for a spot-light)
-    const float3 center = lgtDat.center.xyz;
-    const float radius = lgtDat.radius;
-    const float2 scaleXY = lgtDat.scaleXY;
+float3x3 Invert3x3(float3x3 R)
+{
+    float3x3 C   = transpose(R); // Row to column
+    float    det = dot(C[0], cross(C[1], C[2]));
+    float3x3 adj = float3x3(cross(C[1], C[2]),
+                            cross(C[2], C[0]),
+                            cross(C[0], C[1]));
+    return rcp(det) * adj;
+}
 
-    {
-        if(sideIndex<6 && lgtIndex<(int) g_iNrVisibLights)      // mask 2 out of 8 threads
-        {
-            float3 q0, q1, q2, q3;
-            GetHullQuad(q0, q1, q2, q3, boxX, boxY, boxZ, center, scaleXY, sideIndex);
+float4x4 Homogenize3x3(float3x3 R)
+{
+    float4x4 M = float4x4(float4(R[0], 0),
+                          float4(R[1], 0),
+                          float4(R[2], 0),
+                          float4(0,0,0,1));
+    return M;
+}
 
+float4x4 PerspectiveProjection4x4(float a, float g, float n, float f)
+{
+    float b = (f + n) * rcp(f - n);    // Z in [-1, 1]
+    float c = -2 * f * n * rcp(f - n); // No Z-reversal
 
-            const float4 vP0 = mul(g_mProjection, float4(q0, 1));
-            const float4 vP1 = mul(g_mProjection, float4(q1, 1));
-            const float4 vP2 = mul(g_mProjection, float4(q2, 1));
-            const float4 vP3 = mul(g_mProjection, float4(q3, 1));
+    return float4x4(g/a, 0, 0, 0,
+                      0, g, 0, 0,
+                      0, 0, b, c,
+                      0, 0, 1, 0);
+}
 
-            // test vertices of one quad (of the convex hull) for intersection
-            const unsigned int uFlag0 = GetClip(vP0);
-            const unsigned int uFlag1 = GetClip(vP1);
-            const unsigned int uFlag2 = GetClip(vP2);
-            const unsigned int uFlag3 = GetClip(vP3);
+/* ------------------------------ Implementation ---------------------------- */
+
+#define DUMB_COMPILER // Improve the quality of generated code at the expense of readability
```

Comment:
Unless using an IDE with collapsing of #if, the presence of this makes it a bit harder to follow the code. 

I assume it is left as define here to show the more readable version, but couldn't we get rid of that and instead comment the "dumb compiler" part so that we don't need the other version?

Reply:
The idea is to eventually remove this #define. It's a trivial transformation, and a compiler should be able to do it for us.

We need to report it to Sony and see if it works as expected on Microsoft's machines.